var/spack/repos/builtin/packages/umt/package.py

```diff
@@ -0,0 +1,196 @@
+    depends_on("camp", when="+umpire")
+
+    ####################################################################
+    # Note: cmake, build, and install stages are handled by CMakePackage
+    ####################################################################
+
+    def _get_sys_type(self, spec):
+        sys_type = spec.architecture
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        return sys_type
+
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            hostname = hostname.rstrip("1234567890")
+        return "{0}-{1}-{2}@{3}.cmake".format(
+            hostname,
+            self._get_sys_type(self.spec),
+            self.spec.compiler.name,
+            self.spec.compiler.version,
+        )
+
+    def cmake_args(self):
+        options = []
+        return options
+
+    def _get_host_config_path(self, spec):
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        host_config_path = "{0}-{1}-{2}-{3}.cmake".format(
+            socket.gethostname(), sys_type, spec.compiler, spec.dag_hash()
+        )
+        dest_dir = spec.prefix
+        host_config_path = os.path.abspath(join_path(dest_dir, host_config_path))
+
+        return host_config_path
+
+    # Override the parent class function until its fixed in spack.  (Greg Becker and Chris White are aware of bug)
+    def flag_handler(self, name, flags):
+        if name in ("cflags", "cxxflags", "cppflags", "fflags", "ldflags"):
+            return (None, None, None)  # handled in the cmake cache
+        return (flags, None, None)
+
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+
+        if "umpire" in spec:
+            if "+rocm" in spec["umpire"]:
+                spec.compiler_flags["cxxflags"].append("-I{0}/include".format(spec["hip"].prefix))
+
+        #######################
+        # Note - call the super class AFTER changing any flags, as the super class
+        # adds the cflags, cxxflags, fflags, ldflags, etc, to the cache entries list.
+        # If you try adding any of these yourself you will end up with duplicates.
+        # - black27
+        entries = super(Umt, self).initconfig_compiler_entries()
+
+        if "+openmp" in spec:
+            if spec.satisfies("%xl"):
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_RELEASE", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_Fortran_FLAGS_DEBUG", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_RELEASE", "-qsmp=omp"))
+                entries.append(cmake_cache_string("TETON_OpenMP_CXX_FLAGS_DEBUG", "-qsmp=omp"))
+
+                if "+openmp" in spec:
+                    entries.append(cmake_cache_string("TETON_OpenMP_Offload_Fortran_FLAGS", "-qoffload -qtgtarch=sm_70"))
```

Comment:
This needs to be under the `openmp_offlag` variant, not the openmp variant, and it should have another variant for the NVIDIA architecture (standard in Spack is `cuda_arch`, and you can look at `CUDAPackage` to see how it can be conditional on another variant, although I see you may want a different name since it's fortran offload only).